### DISCUSSION TITLE
Verbose dev server discovery check

### DISCUSSION THREAD

**Discussion by michaelhays** [inngest/inngest#473] [Q&A] [answered]
Hello, I'm running a Next.js app server and Inngest dev server locally, and the "discovery" calls from [`discovery.go`](https://github.com/inngest/inngest/blob/99fc80c190c6415bbb52ae210f1730d6458b7e3c/pkg/devserver/discovery.go#L34-L42) are quite verbose in the terminal.

```sh
$ pnpm next dev --turbo
- info Loaded env from /path/to/project/.env
>>> TURBOPACK (beta)

...

ready - started server on [::]:3000, url: http://localhost:3000
[404] /.netlify/functions/inngest (2724ms)
[404] /.redwood/functions/inngest (746ms)
[404] /x/inngest (4795ms)
event - 4 requests (GET /x/inngest, ...) 6819ms
[404] /x/inngest (98ms)
[404] /.netlify/functions/inngest (87ms)
[404] /.redwood/functions/inngest (88ms)
event - 4 requests (GET /x/inngest, ...) 340ms
```

^ Those last 4 lines repeat every couple of seconds.

```sh
$ pnpm inngest dev
11:00AM INF executor > service starting
...
11:00AM INF api > starting server addr=0.0.0.0:8288


        Inngest dev server online at 0.0.0.0:8288, visible at the following URLs:

         - http://127.0.0.1:8288 (http://localhost:8288)
         - http://192.168.68.108:8288

11:00AM INF devserver > initializing scheduled functions len=0
11:00AM INF devserver > registered functions app=myapp framework=nextjs len=2 sdk=js:v2.1.0 url=http://localhost:3000/api/inngest
11:00AM INF devserver > initializing scheduled functions len=0
11:00AM INF devserver > registered functions app=myapp framework=nextjs len=2 sdk=js:v2.1.0 url=http://localhost:3000/api/inngest
```

^ Those last 2 lines repeat every couple of seconds.

Any tips on how to silence these?

**Comment by darwin67** [comment, marked as answer]
Hey, if you upgrade to the lastest one, it should be trimmed.

**Reply by michaelhays** [reply]
Excellent, thanks for the quick fix! Confirmed that the Inngest dev server is now trimmed.

I'm still seeing these 4 lines repeating every few seconds in my Next.js app server:

```sh
$ pnpm next dev --turbo

[404] /x/inngest (46ms)
[404] /.netlify/functions/inngest (91ms)
[404] /.redwood/functions/inngest (91ms)
event - 5 requests (GET /x/inngest, ...) 325ms
```

Not sure if there's anything you can do there on your side, since I'm guessing you need those pings to run routinely.

I'll probably try to find a way to silence these specific requests on the Next.js side.

**Reply by michaelhays** [reply]
FWIW, this is what I'm doing for the time being on Linux:

```bash
$ pnpm next dev --turbo | grep -v "/inngest"
```

**Reply by darwin67** [reply]
yeah, we can take a look but like you said, we need those periodic pings to automatically register your functions when you develop locally.

the way you're piping works or you should be able disable the automatic discovery completed with an option like `inngest-cli dev --no-discovery`, but that will mean you need to register your functions with the dev server manually.

It's kinda like a tradeoff for convenience.

**Reply by ellisio** [reply]
@darwin67 Is there any chance we could make this configurable? Perhaps a `--discovery-path /api/inngest` flag, or something? If this flag is not provided, default to the current behavior. This would allow us to opt into declarative discovery.
